### PR TITLE
Test Material buttons against a11y contrast guidelines

### PR DESCRIPTION
**Pull request by HansMuller**
Added a11y constrast guideline unit tests for the default configurations of RaisedButton, OutlineButton, FlatButton, and MaterialButton.

Made a few additional gratuitous cosmetic changes to the button tests.

### CODE REVIEW

File: packages/flutter/test/material/flat_button_test.dart [hunk 1]
```diff
@@ -32,6 +32,33 @@ void main() {
...
+            child: FlatButton(
+              child: const Text('FlatButton'),
+              onPressed: () { },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Default, not disabled
```
Comment (johnsonmh) [inline packages/flutter/test/material/flat_button_test.dart]:
```suggestion
    // Default, not disabled.
```

https://www.dartlang.org/guides/language/effective-dart/documentation#do-format-comments-like-sentences

File: packages/flutter/test/material/flat_button_test.dart [hunk 2]
```diff
@@ -32,6 +32,33 @@ void main() {
...
+    await expectLater(tester, meetsGuideline(textContrastGuideline));
+
+    // Highlighted (pressed)
```
Comment (johnsonmh) [inline packages/flutter/test/material/flat_button_test.dart]:
```suggestion
    // Highlighted (pressed).
```

https://www.dartlang.org/guides/language/effective-dart/documentation#do-format-comments-like-sentences

File: packages/flutter/test/material/flat_button_test.dart [hunk 3]
```diff
@@ -32,6 +32,33 @@ void main() {
...
+    final Offset center = tester.getCenter(find.byType(FlatButton));
+    await tester.startGesture(center);
+    await tester.pump(const Duration(milliseconds: 200)); // wait for splash to be well under way
```
Comment (johnsonmh) [inline packages/flutter/test/material/flat_button_test.dart]:
```suggestion
    await tester.pump(const Duration(milliseconds: 200)); // Wait for splash to be well under way.
```

https://www.dartlang.org/guides/language/effective-dart/documentation#do-format-comments-like-sentences

File: packages/flutter/test/material/flat_button_test.dart (hunk 1 above)
Comment (johnsonmh) [reply packages/flutter/test/material/flat_button_test.dart]: (Here and below)

File: packages/flutter/test/material/outline_button_test.dart [hunk 4]
```diff
@@ -412,11 +418,33 @@ void main() {
...
 
     // Tap gesture completes, button returns to its initial configuration.
     await gesture.up();
     await tester.pumpAndSettle();
-    checkPhysicalLayer(buttonElement, fillColor.withAlpha(0x00));
+    _checkPhysicalLayer(buttonElement, fillColor.withAlpha(0x00));
   });
 }
+
+PhysicalModelLayer _findPhysicalLayer(Element element) {
```
Comment (johnsonmh) [inline packages/flutter/test/material/outline_button_test.dart]: these were just renamed and moved right? No functionality change?

**Review by johnsonmh — APPROVED** [review]
LGTM

File: packages/flutter/test/material/outline_button_test.dart (hunk 4 above)
Comment (HansMuller) [reply packages/flutter/test/material/outline_button_test.dart]: That's right. They were jammed at the top of main() with the wrong indent. We usually stick helpers like this at the end of test files and often (not always) use file-private names. That's all I did here.

**Pull request merged**